cadence-style-guide.md

```diff
@@ -0,0 +1,60 @@
+# Cadence Style Guide
+## Table of contents
+1. [Indentation](#indentation)
+1. [Line length](#line-length)
+1. [Comments](#comments)
+1. [Whitespace between code lines](#whitespace-between-code-lines)
+1. [Declaring variables and constants](#whitespace-between-code-lines)
+1. [Bracket spaces](#bracket-spaces)
+1. [Panic messages](#panic-messages)
+1. [Naming and capitalization](#naming-and-capitalization)
+## Indentation
+Use 4 spaces per indentation level.
+Prefer spaces over tabs as indentation method, mixing them should be avoided.
+## Line length
+A line length of 80 characters is recommended for Cadence. This could be particularly challenging when writing capability-related lines, for instance:
```

Comment:
I agree the guide should recommend a reasonable maximum line length, but [we're not writing code on punch cards anymore](https://softwareengineering.stackexchange.com/questions/148677/why-is-80-characters-the-standard-limit-for-code-width). Maybe something like 100?